frontend/src/views/admin/assignments/accept-offer-button-with-dialog.tsx

```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { ActionButton } from "../../../components/action-buttons";
+import { OfferConfirmationDialog } from "./offer-confirmation-dialog";
+import { Assignment } from "../../../api/defs/types";
+import { FaCheck } from "react-icons/fa";
+
+export function AcceptOfferButtonWithDialog(props: {
+    actionButtonEnable: boolean;
+    actionButtonOnClick: () => void;
+    selectedAssignments: Assignment[];
+    dialogVisible: boolean;
+    dialogSetVisible: Function;
+    actionCallback: Function;
+}) {
+    return (
+        <React.Fragment>
+            <ActionButton
+                icon={<FaCheck />}
+                onClick={props.actionButtonOnClick}
```

Comment:
Rather than accessing props all the time, destructure directly in the function arguments.